native/packages/react-native-bpk-component-pagination-dots/src/BpkPaginationDots.js

```diff
@@ -52,136 +49,102 @@ const styles = StyleSheet.create({
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.md,
-    INDICATOR_SIZES.sm,
-  ],
-  middle: [
-    INDICATOR_SIZES.sm,
-    INDICATOR_SIZES.md,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.md,
-    INDICATOR_SIZES.sm,
-  ],
-  closeToEnd: [
-    INDICATOR_SIZES.invisible,
-    INDICATOR_SIZES.sm,
-    INDICATOR_SIZES.md,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.md,
-  ],
-  end: [
-    INDICATOR_SIZES.invisible,
-    INDICATOR_SIZES.sm,
-    INDICATOR_SIZES.md,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-    INDICATOR_SIZES.base,
-  ],
-};
-
-const createIndicatorsFromSizes = (sizesArray, selectedIndex) =>
-  sizesArray.map((indicatorSize, index) => (
-    <BpkPaginationDotsIndicator
-      key={index.toString()}
-      selected={selectedIndex === index}
-      size={indicatorSize}
-    />
-  ));
-
-const getIndicators = (pageCount, selectedIndex) => {
-  if (pageCount <= MAX_DOTS_BEFORE_OVERFLOW) {
-    return new Array(pageCount)
-      .fill()
-      .map((_, index) => (
-        <BpkPaginationDotsIndicator
-          key={index.toString()}
-          selected={selectedIndex === index}
-        />
-      ));
+const getIndicatorSlice = (pageCount, selectedIndex) => {
+  if (pageCount <= 5) {
+    return {
+      begin: 0,
+      end: pageCount,
+    };
   }
 
-  const maxPageIndex = pageCount - 1;
-
   if (selectedIndex <= 2) {
-    return createIndicatorsFromSizes(indicatorMappings.start, selectedIndex);
+    return {
+      begin: 0,
+      end: 5,
+    };
   }
-  if (selectedIndex === 3) {
-    return createIndicatorsFromSizes(
-      indicatorMappings.closeToStart,
-      NUM_OF_DOTS_WHEN_OVERFLOWING - 4,
-    );
+
+  if (selectedIndex >= pageCount - 3) {
+    return {
+      begin: pageCount - 5,
+      end: pageCount,
+    };
   }
-  if (selectedIndex === maxPageIndex - 1) {
-    return createIndicatorsFromSizes(
-      indicatorMappings.closeToEnd,
-      NUM_OF_DOTS_WHEN_OVERFLOWING - 2,
-    );
+
+  return {
+    begin: selectedIndex - 2,
+    end: selectedIndex + 3,
+  };
+};
+
+const getIndicatorSize = (pageCount, index, selectedIndex) => {
+  if (pageCount <= 5 || selectedIndex === index) {
+    return INDICATOR_SIZES.base;
   }
-  if (selectedIndex === maxPageIndex) {
-    return createIndicatorsFromSizes(
-      indicatorMappings.end,
-      NUM_OF_DOTS_WHEN_OVERFLOWING - 1,
-    );
+
+  if (index >= selectedIndex - 1 && index <= selectedIndex + 1) {
+    return INDICATOR_SIZES.md;
   }
-  return createIndicatorsFromSizes(
-    indicatorMappings.middle,
-    NUM_OF_DOTS_WHEN_OVERFLOWING - 3,
-  );
+
+  return INDICATOR_SIZES.sm;
 };
 
 export type Props = {
-  accessibilityLabel: string | ((number, number) => string),
   pageCount: number,
   selectedIndex: number,
-  style: ?any,
+  style: ?StyleObj,
 };
 
+const getIndicatorIndexFromKey = key => parseInt(key.replace(/\D/g, ''), 10);
```

Comment:
I'm not sure we have a convention for using `parseInt` or `Number.parseInt` - I feel that the latter is a bit nicer. Wdyr?

Reply:
![image](https://user-images.githubusercontent.com/1440090/40792514-06ac9062-64f2-11e8-8827-5c2dad650097.png)

Looks like we just use `parseInt` straight up.

Reply:
Fair enough, thanks for confirming!